lib/core/src/animation/animator/easing.rs

```diff
@@ -0,0 +1,88 @@
+//! This file provides the implementation of EasingAnimator.
+
+use super::ContinuousAnimator;
+use crate::animation::position::HasPosition;
+use crate::animation::easing::FnEasing;
+use crate::control::event_loop::EventLoop;
+use crate::math::utils::linear_interpolation;
+
+use nalgebra::Vector3;
+use nalgebra::clamp;
+use std::rc::Rc;
+use std::cell::RefCell;
+
+
+// ==========================
+// === EasingAnimatorData ===
+// ==========================
+
+struct EasingAnimatorData {
+    object              : Box<dyn HasPosition>,
+    origin_position     : Vector3<f32>,
+    target_position     : Vector3<f32>,
+    duration_seconds    : f64,
+    continuous_animator : Option<ContinuousAnimator>
+}
+
+
+
+// ======================
+// === EasingAnimator ===
+// ======================
+
+/// This struct animates from `origin_position` to `target_position` using easing functions.
+pub struct EasingAnimator {
+    data : Rc<RefCell<EasingAnimatorData>>
+}
+
+impl EasingAnimator {
+    /// Creates an EasingAnimator using a `easing_function` to move `object` from
+    /// `origin_position` to `target_position` in `duration_seconds`.
+    pub fn new<T,F>
+    (mut event_loop   : &mut EventLoop
+     , easing_function  : F
```

Comment:
the dots should be aligned with ()
And `where` should be intended one level less.